operator/docs/enhancements/ruler_support.md

```diff
@@ -0,0 +1,844 @@
+
+**Mitigation**: We require a support matrix for `RulerConfig` versions to AlertManager API versions and Remote Write implementations. For later we could use a list of releases (e.g. Prometheus 2.x, Thanos 0.20.z).
+
+#### Two ConfigMaps for all RecordingRule and AlertingRule custom resources
+
+**Risk**: The proposed reconciliation approach manifests that all `AlertingRule` instances are transformed to individual entries in a single ConfigMap. Alerting rules can be become quite big (e.g. large LogQL expresssions). As per ConfigMaps represent Kubernetes resource stored in etcd, this might hit some store limits (See [1MiB limit](https://kubernetes.io/docs/concepts/configuration/configmap/#motivation)). Therefore storing all rules in a single ConfigMap might become impossible over time or on large clusters.
+
+**Mitigation**: Introduce some sort of sharding of `AlertingRule** custom resources into multiple ConfigMap resources.
+
+**Note**: The same applies to `RecordingRule`
+
+## Design Details
+
+### Open Questions [optional]
+
+1.0000 Do we need to add support for Remote Write Configuration per tenant? (See `ruler_remote_write_*` parameters in [limits_config](https://grafana.com/docs/loki/latest/configuration/#limits_config))
+2. Do we need to add support for the `ruler_evaluation_delay_duration`, `ruler_max_rules_per_rule_group` and `ruler_max_rule_groups_per_tenant` limits?
+
+## Implementation History
+
+* 2022-04-21: Initial draft proposal
+* 2022-04-21: Spike implementation for `LokiRule` reconciliation and `RulerConfig` types. (See [PR](https://github.com/grafana/loki/pull/5986))
+* 2022-04-26: Update draft to use `LokiRule` types to use webhook-based validation and `LokiRuleSpec.Selector`, `LokiRuleSpec.NamespaceSelector`. (See [PR](https://github.com/grafana/loki/pull/5986))
+* 2022-04-28: Update draft to split `LokiRule` into two distinct types `AlertingRule` and `RecordingRule`. (See [PR](https://github.com/grafana/loki/pull/5986))
+* 2022-04-28: Renamed `LokiRulerConfig` types to `RulerConfig` as per `Loki` prefix is part of the API group `loki.grafana.com`, i.e. fully qualified type is `ruler.loki.grafana.com`
+
+## Drawbacks
+
+The above proposed design and implementation for LokiStack Ruler support adds a significant amount of new APIs as well as complexity into exposing a declarative approach **only** for the ruler component. Regardless the hard effort to minimize the amount of configuration settings in the proposed CRDs it remains still a huge addition to the operator code base. In contrast to the existing `LokiStack` CRD that is a very slim set of Loki settings (Note: without considering the gateway tenant configuration), the `RulerConfig` is almost one-to-one identical to the [ruler config](https://grafana.com/docs/loki/latest/configuration/#ruler).
```

Comment:
I would consider slimming down some of these specs  and instead hardcoding/using defaults until the need for such complexity is proven. This would reduce the new surface area by a large margin and options could be added incrementally as needed.

Reply:
@owen-d From a user experience perspective the amount of non-defaulted fields the user needs to declare thanks to `+kubebuilder:default` annotation is limited to a minimum. Do you foresee any upcoming changes for `remote_write` or `alertmanager` (e.g. fields that will be deprecated, removed or replaced)?

Reply:
I don't see any changes coming, but @dannykopping may have a better sense than I.